tests/Auth0.AuthenticationApi.IntegrationTests/PasswordlessTests.cs

```diff
@@ -106,5 +115,167 @@ public async Task Can_launch_sms_flow()
                 response.PhoneNumber.Should().Be(request.PhoneNumber);
             }
         }
+
+        [Fact(Skip = "Run manually")]
+        public async Task Can_exchange_sms_code_for_access_token()
+        {
+            var authenticationApiClient = new AuthenticationApiClient(GetVariable("AUTH0_AUTHENTICATION_API_URL"));
+
+            // Exchange the code
+            var token = await authenticationApiClient.GetTokenAsync(new PasswordlessSmsTokenRequest
+            {
+                ClientId = GetVariable("AUTH0_PASSWORDLESSDEMO_CLIENT_ID"),
+                ClientSecret = GetVariable("AUTH0_PASSWORDLESSDEMO_CLIENT_SECRET"),
+                Code = "...",
+                Audience = GetVariable("BRUCKE_MANAGEMENT_API_AUDIENCE"),
+                Scope = "openid email",
+                PhoneNumber = phone
+
+            });
+
+            // Assert
+            token.Should().NotBeNull();
+        }
+
+        [Fact(Skip = "Run manually")]
+        public async Task Can_exchange_email_code_for_access_token()
+        {
+            var authenticationApiClient = new AuthenticationApiClient(GetVariable("AUTH0_AUTHENTICATION_API_URL"));
+
+            // Exchange the code
+            var token = await authenticationApiClient.GetTokenAsync(new PasswordlessEmailTokenRequest
+            {
+                ClientId = GetVariable("AUTH0_PASSWORDLESSDEMO_CLIENT_ID"),
+                ClientSecret = GetVariable("AUTH0_PASSWORDLESSDEMO_CLIENT_SECRET"),
+                Code = "...",
+                Audience = GetVariable("BRUCKE_MANAGEMENT_API_AUDIENCE"),
+                Scope = "openid email",
+                Email = email
+
+            });
+
+            // Assert
+            token.Should().NotBeNull();
+        }
+
+        [Fact]
+        public async Task Should_Request_a_token_using_sms_code()
```

Comment:
Are the mocks in this test set up as a spy or is the request to the server fully mocked? If it's the latter, as these are integration tests, what value do we get by mocking the server transaction? Isn't the goal to integrate with the server in the same way that the other tests do?

Same question for the passwordless with email test.

Reply:
I couldn't find a way to integration test this as I have no way to get a code to send as part of this request (it requires email or phone access).